### DISCUSSION TITLE
Error while installing PimcoreDataHubBundle for Data Import

### DISCUSSION THREAD

**Discussion by abdullah-sediq** [pimcore/pimcore#15493] [Q&A / Support]
I urgently need help for my study project because I have been working with Pimcore and have been fascinated by it. Therefore, I have chosen Pimcore for my study project. I want to import data from third-party systems into Pimcore. I imagined doing this through Pimcore's DataHub. However, when I try to install the DataHub and Dataimport Bundle, I encounter the following error messages. Can someone please help me with this?

When executing the command:  "bin/console pimcore:bundle:install PimcoreDataHubBundle," I receive the following error messages:

In Kernel.php line 219:

  Bundle "PimcoreDataHubBundle" does not exist or it is not enabled. Maybe yo
  u forgot to add it in the "registerBundles()" method of your "App\Kernel.ph
  p" file?

pimcore:bundle:install [-f|--fail-without-error] [--no-post-change-commands] [--no-assets-install] [--no-cache-clear] [--] <bundle>

How can I resolve this issue?

**Comment by kingjia90** [comment]
if you are on Pimcore 10.5, please run `bin/console pimcore:bundle:enable PimcoreDataHubBundle` first
if you are on Pimcore 11, you have to add this line first https://github.com/pimcore/demo/blob/56ce2047b4c975ab87773528837d5c9ecf7bff36/config/bundles.php#L33

**Reply by abdullah-sediq** [reply]
Thank you for your feedback. However, I am facing an issue with the next step, even though I have configured the database correctly. When executing the command "bin/console pimcore:bundle:install PimcoreDataHubBundle," I receive the following error message:

[ERROR] An exception occurred in the driver: SQLSTATE[HY000] [2002]
php_network_getaddresses: getaddrinfo for db failed: nodename nor
servname provided, or not known

**Reply by kingjia90** [reply]
Looks like there some db setup/server configuration error?  did you run `php vendor/bin/pimcore-install`?

**Reply by StefanRickli** [reply]
Just had this problem yesterday. The issue is that pimcore installation would reference the DB server service `db` by its docker internal hostname `db` in `config/local/database.yaml`. `db` can only be resolved from within the docker containers.

Either replace `db` with `localhost`, but then you have to also add a port export in the `docker-compose.yaml`.

Or, use the `php` container's `php` and `composer` commands. Like so:
- `docker container list`: look up the `php` container's name
- `docker exec php_container_name composer require pimcore/data-hub`
- `docker exec php_container_name php /var/www/html/bin/console pimcore:bundle:list`
- Now add the line in `bundles.php`
- `docker exec php_container_name php /var/www/html/bin/console pimcore:bundle:install PimcoreDataHubBundle`

Hope this helps
